collector/src/test/java/com/google/cloud/trace/zipkin/ZipkinCollectorIntegrationTest.java

```diff
@@ -0,0 +1,102 @@
+package com.google.cloud.trace.zipkin;
+
+import com.google.auth.Credentials;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.common.collect.Lists;
+import org.hamcrest.Matchers;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.ProvideSystemProperty;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+import org.springframework.core.task.SyncTaskExecutor;
+import org.springframework.test.context.junit4.SpringRunner;
+import zipkin.Codec;
+import zipkin.Span;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Executor;
+
+import static com.google.cloud.trace.zipkin.StackdriverZipkinCollector.ZIPKIN_CONFIG_NAMES;
+import static java.lang.Math.min;
+import static java.util.Arrays.asList;
+import static java.util.Collections.unmodifiableList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.hasSize;
+import static org.mockito.Mockito.mock;
+import static zipkin.TestObjects.LOTS_OF_SPANS;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = {StackdriverZipkinCollector.class, ZipkinCollectorIntegrationTest.TestConfiguration.class},
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    properties = {ZIPKIN_CONFIG_NAMES, "zipkin.storage.stackdriver.project-id=test_project"})
+public class ZipkinCollectorIntegrationTest
+{
+  //LOTS_OF_SPANS is too much! I need at most 30!
+  private static final List<Span> TEST_SPANS = unmodifiableList(asList(Arrays.copyOfRange(LOTS_OF_SPANS, 0, min(30, LOTS_OF_SPANS.length))));
+
+  @ClassRule
+  public static final StackdriverTraceClassRule traceService = new StackdriverTraceClassRule();
+
+  @ClassRule
+  public static final ProvideSystemProperty hostProperty
```

Comment:
You can't use `@SpringBootTest` property because you need this to be set before the context is started? Cause in general if in the code you're referencing that `zipkin.storage.stackdriver.api-host` property via `@Value` then it's enough to just override it as a `property` in `@SpringBootTest`.